openedx/openedx-learning
# Ensure the correct Taxonomy subclass is used

## Description

Ensures that we always use the correct Taxonomy model subclass whenever returning or using Taxonomy instances.

## Supporting information

Proposed way to address https://github.com/openedx/edx-platform/pull/32518#discussion_r1244335301

## Testing instructions

Manual testing is possible but not very useful, since we don't yet have views for these models and APIs.

But you can check that the added tests cover the expected polymorphism behavior supported by this change, and that tests are still covering 100% of the `openedx_tagging` module.

## Other information

Here we add a basic SystemTaxonomy class to help validate this change, but more functionality for system taxonomies will be added by a subsequent PR.

Author to do before merge:

* Address [ObjectTag.taxonomy issue](https://github.com/openedx/openedx-learning/pull/60#discussion_r1246098521)
